src/radical/pilot/staging_directives.py

```diff
@@ -141,17 +141,21 @@ def complete_url(path, context, log=None):
     certain locations, namely relative to
 
         * `client://`  : the client's working directory
-        * `resource://`: the RP    sandbox on the target resource
-        * `pilot://`   : the pilot sandbox on the target resource
-        * `task://`    : the task  sandbox on the target resource
+        * `endpoint://`: the root file system on the target resource
+        * `resource://`: the sandbox base dir on the target resource
+        * `session://` : the session sandbox  on the target resource
+        * `pilot://`   : the pilot   sandbox  on the target resource
+        * `task://`    : the task    sandbox  on the target resource
+
+    All location are interpreted as directories, never as files.
 
     For the above schemas, we interpret `schema://` the same as `schema:///`,
     ie. we treat this as a namespace, not as location qualified by a hostname.
 
     The `context` parameter is expected to be a dict which provides a set of
-    URLs to be used to expand the path.
+    URLs to be used to expand the path for those schemas
 
-    Other URL schemas are left alone, any other strings are interpreted as
+    Other URL schemas are left alone, any non-URL strings are interpreted as
     path in the context of `pwd`.
```

Comment:
This is incomplete, and it is not clear what `pwd` refers to. I suggest making a change similar to the following, depending on additional documentation expansion.
```suggestion
    Other URL schemes are left alone. Any non-URL strings are interpreted as
    paths. Absolute paths are interpreted as ``file:///path``. Relative paths are
    interpreted as ``pwd:///path``.
```

Reply:
This has been applied - thanks.